### ISSUE TITLE
WireViz 0.4 changed CLI parameters 

### ISSUE THREAD

**Issue by martinrieder** [nanangp/vscode-wireviz-preview#5]
A WireViz "release candidate" [release/v0.4-rc2 was merged into master](https://github.com/wireviz/WireViz/commit/954c4f5f92b1d0531f6af0c318c60786b7906d42) based on PR wireviz/WireViz#339. See the [change log here](https://github.com/wireviz/WireViz/blob/v0.4/docs/CHANGELOG.md) for details about new command line options and output file formats. Note that also [examples 11-14](https://github.com/wireviz/WireViz/blob/master/examples/readme.md#example-11) were added to demonstrate the new features.

The VSCode extension is still able to run WireViz 0.4 and show an output image. However, it would not show an error messages to the user when it fails and therefore no files were generated. Additionally, if the picture already existed from a previously successful run, this would be presented to the user instead without any error indication.

I am pasting the `--help` message here for reference:

```
Usage: wireviz [OPTIONS] [FILE]...

  Parses the provided FILE and generates the specified outputs.

Options:
  -f, --format TEXT       Output formats (see below).  [default: hpst]
  -p, --prepend PATH      YAML file to prepend to the input file (optional).
  -o, --output-dir PATH   Directory to use for output files, if different from
                          input file directory.
  -O, --output-name TEXT  File name (without extension) to use for output
                          files, if different from input file name.
  -V, --version           Output WireViz version and exit.
  --help                  Show this message and exit.

  The -f or --format option accepts a string containing one or more of the
  following characters to specify which file types to output: c (CSV), g (GV),
  h (HTML), p (PNG), P (PDF), s (SVG), t (TSV)
```

**Comment by nanangp** [comment]
Oh nice. Looking at the changelog, seems like this should resolve some of the clutter that resulted in workarounds for #3. 

I'll also have a look at why errors are being swallowed. Changelog didn't seem to mention anything.

PS. Sorry turnaround may be a little slow on this. I've got other stuff on my plate at the moment. If we're lucky, we may have something by the time 0.4.0 reaches RTM.

**Comment by nanangp** [comment]
Currently blocked by wireviz/WireViz#344 on 0.4, so I couldn't test anything. Will probably hold off until their proposed fix (planned for 0.4.1) has been properly tested and released.

**Comment by nanangp** [comment]
@martinrieder would you be willing to test the `wireviz0.4` branch of this extension? 

I've made it work with their proposed fix for the aforementioned issue, but I'd like it for someone running an official 0.4 to confirm.

**Comment by martinrieder** [comment]
Would it be sufficient to only replace the files for testing? Or do I need to reinstall the extension?

**Comment by nanangp** [comment]
If you're running the extension from source, you could just git pull/checkout the whole branch and run it.

**Comment by martinrieder** [comment]
@nanangp Thanks a lot for this quick reaction.

No, I had it installed from Marketplace, so I did the `git checkout` and `npm run build` separately, then copied over the `extension.js` file. 

I can confirm that it exectes, though there are some timing issues.  When VScode has not finished saving the file or you double click the Prieview (F8) button, you _might_ see an Error shortly before it displays and image. I have not been able to capture a screenshot of it yet.

Besides these glitches, I am quite happy about your changes. Pictures are generated for both PNG and SVG, even with included images. These can also be placed in another folder!

**Comment by martinrieder** [comment]
Could you possibly feed to the output messages to the console instead of the panel? Or STDOUT to the console and STDERR to the panel?

**Comment by nanangp** [comment]
> though there are some timing issues.

That's interesting. It's possible that `process.spawn` hasn't finished when the extension is triggered the second time. 

This may have been an existing bug, or a side effect of the small change to using the process' `close` event instead of `exit`.  I'll see if this is a quick fix. If not, we'll open a new issue and fix it later. 

>Could you possibly feed to the output messages to the console instead of the panel?

Sounds good. I'll add debug console logging *in addition* to the panel.

It's rather annoying that WV is using stdout instead of stderr for a bunch of their errors. This was meant to be fixed in the stdout PR that is yet to be merged.

**Comment by martinrieder** [comment]
> I agree that when "-" as input file should be interpreted as STDIN, then "-" as output file
> similarly should be interpreted as STDOUT. The output filename stem is by default equal to the input filename stem unless specified differently using `-o`.
> 
> I'm not sure if it makes sense to output more than one file format to STDOUT, but try it out to see if it can be usable.
> 
> There are a few places in the code where some warning text is written to STDOUT. Please make sure they are all written to STDERR to avoid possible conflicts with an output file written to STDOUT.

@kvid in https://github.com/wireviz/WireViz/issues/320#issuecomment-1635069465

**Comment by martinrieder** [comment]
Closing this, as we have #8 to follow up.

**Comment by martinrieder** [comment]
Reopened because I found an absolute path in `.vscode/launch.json` from
https://github.com/nanangp/vscode-wireviz-preview/commit/3c5ebbc4125e9bc0c2678c339e81c37baf5f1ba8

**Comment by nanangp** [comment]
> absolute path

Oops
